### PR TITLE
RDK-36282: Add support for passing user scripts

### PR DESCRIPTION
**Pull request by Scony**
allows injecting user scripts via config:
```
"userscripts": ["/tmp/data/script.js", "script2.js", "nonexistent"]
```

### CODE REVIEW

**Review by emutavchi — APPROVED** [review]
looks good to me. let's wait for a review from Metrological team.

File: WebKitBrowser/WebKitImplementation.cpp
Comment (MFransen69) [inline WebKitBrowser/WebKitImplementation.cpp:2270]: Hi, for the error message above it is consistent with other error message in this file to use SYSLOG however for this line  where it is informational I would propose to use "TRACE(Trace::Information, ..." here so we do not pollute the syslog output with informational messages

File: WebKitBrowser/WebKitImplementation.cpp
Comment (MFransen69) [inline WebKitBrowser/WebKitImplementation.cpp:2258]: I guess you do not always expect these scripts to be in the datapath? As that should be the case for a stack where datapath is filled could we (like for the extension path) make these relative to the datapath when datapath is set (or additionally check if the scriptpath is a relative path, does not start with a /, and only then append it the the datapath)?

Comment (MFransen69) [inline WebKitBrowser/WebKitImplementation.cpp:2258]: I guess you do not always expect these scripts to be in the datapath? As that should be the case for a stack where datapath is filled could we (like for the extension path) make these relative to the datapath when datapath is set (or additionally check if the scriptpath is a relative path, does not start with a /, and only then append it the the datapath)?

File: WebKitBrowser/WebKitImplementation.cpp
Comment (MFransen69) [inline WebKitBrowser/WebKitImplementation.cpp:2447]: As the code is similar to the above put it in a private function and call it in both places perhaps?

Comment (MFransen69) [inline WebKitBrowser/WebKitImplementation.cpp:2447]: As the code is similar to the above put it in a private function and call it in both places perhaps?

**Review by MFransen69 — CHANGES_REQUESTED** [review]

**Comment by Firdose2109** [comment]
Can one of the admins verify this patch?

**Comment by Scony** [comment]
@MFransen69 could you please re-review?

**Pull request merged**
